# Sustainable retirement age
AIM: assuming that the current situation of working / retired people is sustainable, and keeping that as reference, what is the retirement age for the whole century, considering the forecast model of the population I have built in my previous Notebook? 

In [1]:
from istatapi import discovery, retrieval
import numpy as np
import pandas as pd
pd.options.plotting.backend = "plotly"
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go

In [2]:
dfpproj = (
    pd.read_csv("data/3_projection.csv", index_col=0)
    .rename(columns={ str(y): y for y in range(2002, 2100+1)})
)
dfpproj

2002    2003    2004    2005    2006    2007    2008    2009    2010  \
eta                                                                           
0    523007  535656  542323  558500  551127  556884  560700  570791  562322   
1    529233  526577  538020  545596  559067  552870  560028  563021  569963   
2    528131  533126  531043  540860  548487  560347  556146  562640  564317   
3    518790  530847  537507  535545  543768  550808  563000  559193  564536   
4    515957  522484  534064  542329  540131  545782  554754  565127  560899   
..      ...     ...     ...     ...     ...     ...     ...     ...     ...   
96    16400   17254   17494   20055   21122   24346   25054   27290   28086   
97    10981   11797   11980   12758   14071   15356   17559   18179   20203   
98     6838    7643    7901    8505    8725    9888   10871   12349   12977   
99     4866    4702    4971    5411    5682    5945    6643    7469    8547   
100    6121    7093    7200    8095    8538    9122    9575   10158   11297   

       2011  ...           2091           2092           2093           2094  \
eta          ...                                                               
0    553218  ...  160985.707782  158609.773171  156285.205876  154011.330260   
1    561694  ...  163420.110773  160985.707782  158609.773171  156285.205876   
2    569064  ...  165922.371868  163420.110773  160985.707782  158609.773171   
3    565219  ...  168491.060255  165922.371868  163420.110773  160985.707782   
4    565563  ...  171130.395538  168491.060255  165922.371868  163420.110773   
..      ...  ...            ...            ...            ...            ...   
96    29780  ...  213040.166208  214043.949284  218152.070513  220923.178559   
97    21305  ...  184372.801995  184233.085880  185457.967691  189381.119293   
98    14740  ...  161366.867065  157725.144522  157939.413665  159325.491959   
99     9207  ...  131742.658587  134993.335698  132234.587381  132702.395794   
100   13106  ...  226975.126137  229332.766911  232913.425873  233434.053115   

              2095           2096           2097           2098  \
eta                                                               
0    151792.681188  149622.656834  147501.870694  145431.575822   
1    154011.330260  151792.681188  149622.656834  147501.870694   
2    156285.205876  154011.330260  151792.681188  149622.656834   
3    158609.773171  156285.205876  154011.330260  151792.681188   
4    160985.707782  158609.773171  156285.205876  154011.330260   
..             ...            ...            ...            ...   
96   225050.799372  226322.555749  231557.021618  229577.856950   
97   192155.057737  196120.367661  197605.937659  202562.260770   
98   163038.959923  165775.199431  169551.464020  171193.800944   
99   134157.690783  137582.034435  140193.502021  143696.384984   
100  234061.113590  235387.446833  238419.440276  242022.005667   

              2099           2100  
eta                                
0    143414.475595  141449.018955  
1    145431.575822  143414.475595  
2    147501.870694  145431.575822  
3    149622.656834  147501.870694  
4    151792.681188  149622.656834  
..             ...            ...  
96   228988.996263  232480.835614  
97   201213.645805  201079.282456  
98   175854.659252  175048.414745  
99   145400.631631  149680.110123  
100  246558.953785  250543.277462  

[101 rows x 99 columns]

In [3]:
# I'm making some assumptions on the average age of retirement and start working
working_age = { y: 20 + 2*(y-2002)/(2022-2002) for y in range(2002,2022+1) } # Assumption: it was 20 in 2002 and it is 22 in 2022
retirem_age = { y: 58 + 7*(y-2002)/(2022-2002) for y in range(2002,2022+1) } # Assumption: it was 58 in 2002 and it is 65 in 2022

# ratio of people in working age over retired age
wr_df = pd.DataFrame(index=range(2002,2022+1))
wr_df["working-age / retirement-age ratio"] = np.nan
for y in wr_df.index:
    wa = working_age[y]
    ra = retirem_age[y]
    wr_df.loc[y, "working-age / retirement-age ratio"] = dfpproj.loc[wa:ra,y].sum() / dfpproj.loc[ra:,y].sum()
    
wr_df

working-age / retirement-age ratio
2002                            2.021864
2003                            2.048581
2004                            2.052414
2005                            2.177448
2006                            2.165623
2007                            2.113820
2008                            2.246041
2009                            2.207381
2010                            2.163356
2011                            2.290755
2012                            2.251336
2013                            2.182787
2014                            2.313530
2015                            2.280575
2016                            2.255003
2017                            2.384517
2018                            2.355673
2019                            2.319199
2020                            2.453503
2021                            2.411512
2022                            2.434195

Note that I'm not consideting unemployment, assuming it is constant, and what matters is the ratio between working and retired people not if working people are actually employed or not.

I will now compute the projected retirement age that it is necessary to maintain an average working/retirement ratio of 2.2

In [13]:
wr_ratio = 2.2

# Same data plotted in two variants
raproj_df = pd.DataFrame(index=range(2002,2100+1))
raproj_df["Sustainable retirement age"] = np.nan
for y in raproj_df.index:
    wa = 22 # Assumption: people will start working at 22 on average
    for ra in range(50, 100): # I'm testing different retirement ages (ra) untill the ratio is 2.2 is reached
        if dfpproj.loc[wa:ra,y].sum() / dfpproj.loc[ra:,y].sum() > wr_ratio:
            raproj_df.loc[y, "Sustainable retirement age"] = ra
            break
    
fig = raproj_df.plot()
fig.update_layout(
    showlegend=False,
    xaxis_title="Year",
    yaxis_title="Retirement Age",
    title="Sustainable retirement age to maintain the same ratio of working/retired people of 2.2"
)
fig.show()

# Variant #2: year of birth on the x-axis
raproj_df["Born year retiring"] = raproj_df.index - raproj_df["Sustainable retirement age"]
fig = px.line(
    data_frame=raproj_df.reset_index().rename(columns={"index": "Year of Retirement"}),
    x="Born year retiring",
    y="Sustainable retirement age",
    hover_data=["Year of Retirement"],
) 
fig.update_layout(
    showlegend=False,
    xaxis_title="Year of Birth",
    yaxis_title="Retirement Age",
    title="Sustainable retirement age to maintain the same ratio of working/retired people of 2.2"
)
fig.show()

## Conclusions
- I made some assumptions on the age at which people starts to work and retired in the past
- I computed a ratio between working-age and retired-age people of 2.0 - 2.4 in the historical data, and picked an average value of 2.2 as the "referance" one to keep constant in the future
- Based on my projection of Italian future population, being born in 1990, I expect to retire at 76 year old in 2066

## Follow-up
- Consider unemployment: this is a self-balancing secondary effect as less woking people might compensate lower unemployment too
- Run a sensitivity test on the "reference" ratio of working-age/retirement-age people as it is a very handwavy estimate
- Get more data on effective historical start age and retirement age
- Run a sensitivity test on the age at which people will start to work
- Compute the life expectancy after retirement: is it fair w.r.t. those who retired in the past years?